### ISSUE TITLE
[Bug]: context page is unresponsive and never existed

### ISSUE THREAD

**Issue by Praveenkumar-95** [microsoft/playwright#33923]
### Version

1.49.0

### Environment

```shell
Playwright Version: playwright==1.49.0
Operating System: Windows 11
Python Version: 3.12
Browser: Chromium 131.0.6778.33
headless= True
```

I was checking if a progress bar was not visible within a 20-minute window. I used statements like expect(locator).not_to_be_visible, locator.is_visible, etc., to return True/False values and proceed with the next actions.

Here's how my code looked:
```
def wait_until_not_visible(page, locator, timeout=1200, interval=5):
    """
    Waits until the element specified by the locator is not visible.

    Args:
        page: The Playwright page object.
        locator: The locator string for the element to check.
        timeout: Maximum time to wait in seconds (default is 1200 seconds = 20 minutes).
        interval: Time in seconds between each check (default is 5 seconds).

    Returns:
        True if the element becomes hidden within the timeout, False otherwise.
    """
    start_time = time.time()
    while time.time() - start_time < timeout:
        log.info(f"Polling condition at {time.time() - start_time:.2f}s")
        try:
            # is_visible = page.locator(locator).is_visible(timeout=100)
            is_visible = page.locator(locator).all()
            is_visible = True if is_visible else False

            if not is_visible:
                return True
        except TimeoutError as e:
            print(f"Error checking visibility: {e}")
            return False

        time.sleep(interval)

    return False
```

```
wait_until_not_visible(context.page,locator,1200,5)
log.info("Upload done")
page.reload() or even tried verifying the other element's visibility.
```

while the execution logs looked like this

```
2024-12-10 12:40:31,051 [INFO ]  [<Locator frame=<Frame name= url='https://10.91.114.149/preferences/updates'> selector="//div[contains(@class,'mantine-Progress-root')] >> nth=0">]
2024-12-10 12:40:31,051 [INFO ]  Polling... Locator visibility: True
2024-12-10 12:40:36,052 [INFO ]  Polling condition at 213.68s
2024-12-10 12:41:14,515 [INFO ]  [<Locator frame=<Frame name= url='https://10.91.114.149/preferences/updates'> selector="//div[contains(@class,'mantine-Progress-root')] >> nth=0">]
2024-12-10 12:41:14,515 [INFO ]  Polling... Locator visibility: True
2024-12-10 12:41:19,516 [INFO ]  Polling condition at 257.14s
2024-12-10 12:42:51,715 [INFO ]  [<Locator frame=<Frame name= url='https://10.91.114.149/preferences/updates'> selector="//div[contains(@class,'mantine-Progress-root')] >> nth=0">]
2024-12-10 12:42:51,715 [INFO ]  Polling... Locator visibility: True
2024-12-10 **12:42:56**,715 [INFO ]  Polling condition at 354.34s
2024-12-10 **12:45:20**,484 [INFO ]  []
2024-12-10 12:45:20,484 [INFO ]  Polling... Locator visibility: False
2024-12-10 12:45:20,484 [INFO ]  Upload done
```

Problem:

- In step 1, the check for visibility took a long time to return a result. Sometimes it worked, sometimes it didn’t. So, I resorted to using a while loop to handle iterative verification and still an issue to identify the locator, take longer time
- Step 2 (log.info("done")) logged successfully, but:
- Step 3 (page.reload() or element visibility check , used all() method ) never occurred or took lot of time .
- from the logs, There was an event where it returned an empty list and it took almost 3 min from the earlier log statement and it was also 5 - 7 minutes sometimes
- and I did a page reload after the log which never occurred, nor did throwing timeout or page is closing

I used pdb.set_trace() after step 2, and printed the page content. It took a lot of time to print, and when I tried page.reload(), it never completed, although I could do it manually in the browser.

If I left the code running, it would just hang there overnight without throwing any error or closing the page.

### Expected behavior

The page should be reloaded or elements should be interactable 

### Actual behavior

The chromium can operate manually but the context page is completely unresponsive

**Comment by yury-s** [comment]
Don't use `time.sleep()`, see [this doc](https://playwright.dev/python/docs/library#timesleep-leads-to-outdated-state), it might be the reason of the failure. Record a trace and try to see what may cause the delay. If you still think this is a bug in Playwright, please share a complete minimal reproduction example which we could run locally to see the problem.

**Comment by Praveenkumar-95** [comment]
I have removed time.sleep as suggested. and I still face the issue

I have been using the BDD framework for the past 2 years and never faced something like this... 

This was not the case with Codegen. It is working fine with Codegen. Do you have any idea what the issue could be?
FYI I am using video recording and the trace as well. 
```

        if context.user.sees_text(text="Uploading software package...", max_time=5000):
            log.info(f"Saw text 'Uploading software package...'")
            # start = datetime.datetime.now().timestamp()
            log.info(f"Uploading of SW package started ")
            # confirm_visibility(context.page,_LOCATORS[context.ui_type]["PROGRESS_BAR"],condition=True,max_time=5000,error=True,)
            # confirm_visibility(context.page,_LOCATORS[context.ui_type]["PROGRESS_BAR"],condition=False,max_time=1200000,error=True,)
            # expect(context.page.get_by_text("Uploading software package...")).to_be_visible()
            expect(context.page.locator(".mantine-Progress-root")).not_to_be_visible(timeout=1200000)
            expect(context.page.get_by_role("heading", name="Installation details")).to_be_visible()
            # end = datetime.datetime.now().timestamp()
            log.info(f"Importing of SW package finished")
            # log.info(f"Total time taken to upload & import the SW package is {(end - start) / 60:.2f} min")
        else:
            log.error(f"Did not see 'Uploading software package...'")
            raise TimeoutError(f"Did not see 'Uploading software package...'")
```

**Comment by yury-s** [comment]
It could be that the behavior of the page has changed, something in your environment, a change in the browser or playwright, unfortunately we can't tell for sure based on the provided information. Try debugging using the playwright tools and if you think this is a playwright bug, please file a new issue with a minimal repro that we could run locally to reproduce it.

**Comment by Praveenkumar-95** [comment]
@yury-s 

below are the playwright api logs
```
2024-12-16T07:06:02.390Z pw:api   locator resolved to <div class="ac-manual-upload-dialog-waiting-response">…</div>
2024-12-16T07:06:02.390Z pw:api   unexpected value "visible"
2024-12-16T07:06:03.395Z pw:api   locator resolved to <div class="ac-manual-upload-dialog-waiting-response">…</div>
2024-12-16T07:06:03.395Z pw:api   unexpected value "visible"
2024-12-16T07:06:04.404Z pw:api   locator resolved to <div class="ac-manual-upload-dialog-waiting-response">…</div>
2024-12-16T07:06:04.404Z pw:api   unexpected value "visible"
2024-12-16T07:06:05.409Z pw:api   locator resolved to <div class="ac-manual-upload-dialog-waiting-response">…</div>
2024-12-16T07:06:05.409Z pw:api   unexpected value "visible"
2024-12-16T07:06:06.412Z pw:api   locator resolved to <div class="ac-manual-upload-dialog-waiting-response">…</div>
2024-12-16T07:06:06.412Z pw:api   unexpected value "visible"

- [ ] 2024-12-16T07:06:10.979Z pw:api   locator resolved to <div class="ac-manual-upload-dialog-waiting-response">…</div>
- [ ] 2024-12-16T07:06:10.979Z pw:api   unexpected value "visible"
- [ ] 2024-12-16T07:06:16.158Z pw:api   locator resolved to <div class="ac-manual-upload-dialog-waiting-response">…</div>
- [ ] 2024-12-16T07:06:16.158Z pw:api   unexpected value "visible"
- [ ] 2024-12-16T07:06:23.381Z pw:api   locator resolved to <div class="ac-manual-upload-dialog-waiting-response">…</div>
- [ ] 2024-12-16T07:06:23.381Z pw:api   unexpected value "visible"
- [ ] 2024-12-16T07:06:37.088Z pw:api   locator resolved to <div class="ac-manual-upload-dialog-waiting-response">…</div>
- [ ] 2024-12-16T07:06:37.088Z pw:api   unexpected value "visible"
- [ ] 2024-12-16T07:06:56.272Z pw:api   locator resolved to <div class="ac-manual-upload-dialog-waiting-response">…</div>
- [ ] 2024-12-16T07:06:56.272Z pw:api   unexpected value "visible"
- [ ] 2024-12-16T07:07:25.107Z pw:api   locator resolved to <div class="ac-manual-upload-dialog-waiting-response">…</div>
- [ ] 2024-12-16T07:07:25.107Z pw:api   unexpected value "visible"
- [ ] 2024-12-16T07:08:21.965Z pw:api   locator resolved to <div class="ac-manual-upload-dialog-waiting-response">…</div>
- [ ] 2024-12-16T07:08:21.965Z pw:api   unexpected value "visible"
- [ ] 2024-12-16T07:10:24.412Z pw:api   locator resolved to <div class="ac-manual-upload-dialog-waiting-response">…</div>
- [ ] 2024-12-16T07:10:24.412Z pw:api   unexpected value "visible"

```



If you notice, I am verifying that a locater will not be visible in a couple of **minutes** using the `expect` condition.

As per the logs playwright API is verifying every 1 sec and suddenly the verification of the element is taking quite a long time
